src/data/matrix/basic.lean

```diff
@@ -644,6 +644,14 @@ lemma smul_mul_vec_assoc (A : matrix m n α) (b : n → α) (a : α) :
   (a • A).mul_vec b = a • (A.mul_vec b) :=
 by { ext, apply smul_dot_product }
 
+lemma mul_vec_add_distrib (A : matrix m n α) (x y : n → α) :
+  A.mul_vec (x + y) = A.mul_vec x + A.mul_vec y :=
+by { ext, apply dot_product_add }
+
+lemma vec_mul_add_distrib (A : matrix m n α) (x y : m → α) :
+  vec_mul (x + y) A = vec_mul x A + vec_mul y A :=
+by { ext, apply add_dot_product }
+
```

Comment:
As mentioned before, I think a better set of names would be
```suggestion
lemma mul_vec_add (A : matrix m n α) (x y : n → α) :
  A.mul_vec (x + y) = A.mul_vec x + A.mul_vec y :=
by { ext, apply dot_product_add }

lemma add_mul_vec (A : matrix m n α) (x y : m → α) :
  vec_mul (x + y) A = vec_mul x A + vec_mul y A :=
by { ext, apply add_dot_product }

```

Reply:
Oh I am so sorry, didn't see that comment. Will change it right away!